docs/2.4/architecture.md

```diff
@@ -0,0 +1,421 @@
+to authenticate when joining the cluster. Teleport does not allow SSH sessions into nodes 
+that are not cluster members.
+
+A node certificate contains the node's role (like `proxy`, `auth` or `node`) as
+a certificate extension (opaque signed string). All nodes in the cluster can
+connect to auth server's HTTP API via SSH tunnel that checks each connecting
+client's certificate and role to enforce access control (e.g. client connection
+using node's certificate won't be able to add and delete users, and can only
+get auth servers registered in the cluster).
+
+### User Certificates
+
+When an auth server generates a user certificate, it uses the information provided
+by the cluster administrator about the role assigned to this user.
+
+A user role can restrict user logins to specific OS logins or to a subset of 
+cluster nodes (or any other restrictions enforced by the role). Teleport's user name is stored in a certificate's key id field. User's certificates do not use any cert extensions as a workaround to the 
+[bug](https://bugzilla.mindrot.org/show_bug.cgi?id=2387) that treats any extension 
+as a critical one, breaking access to the cluster.
+    
+## Audit Log
+
+The Teleport auth server keeps the audit log of SSH-related events that take
+place on any node with a Teleport cluster. It is important to understand that
+the SSH nodes emit audit events and submit them to the auth server. 
+
+!!! warning "Compatibility Warning":
+    Because all SSH events like `exec` or `session_start` are reported by the
+    Teleport node service, they will not be logged if you are using OpenSSH
+    `sshd` daemon on your nodes.
+
+Only an SSH server can report what's happening to the Teleport auth server.
+The audit log is a JSON file which is by default stored on the auth server's
+filesystem under `/var/lib/teleport/log`. The format of the file is documented
+in the [Admin Manual](admin-guide/#audit-log).
+
+Teleport users are encouraged to export the events into an external long term
+storage.
+
+!!! info "Deployment Considerations":
+    If multiple Teleport auth servers are used to service the same cluster (HA mode)
+    a network file system must be used for `/var/lib/teleport/log` to allow them
+    to combine all audit events into the same audit log.
+
+### Session Recording
+
+By default, destination nodes submit SSH session traffic to the auth server
+for storage. These recorded sessions can be replayed later via `tsh play` 
+command or in a web browser.
+
+Some Teleport users mistakenly believe that audit and session recording happen
+on the Teleport proxy server. This is not the case because a proxy cannot see
+the encrypted traffic, it is encrypted end-to-end, i.e. from an SSH client to
+an SSH server/node, see the diagram below:
+
+![session-recording-diagram](img/session-recording.svg)
+
+However, starting from Teleport 2.4 it is now possible to configure the
```

Comment:
Great explanation of the recording proxy!

One advantage of recording sessions at the proxy is that they can be configured for restricted access (while nodes typically have more open access). When sessions are recorded at the nodes a user with legitimate access to the node can add iptables rules to prevent sessions logs from reaching the Auth Server. With sessions recorded at the proxy this is harder (because you don't need to provide everyone SSH access the proxy).